pages/Events.tsx

```diff
@@ -0,0 +1,111 @@
+// Style
+import styles from '../styles/Events.module.scss';
+// Internal Components
+import NavBar from '../components/NavBar';
+import Footer from '../components/Footer';
+import Section from '../layouts/Section';
+import EventBanner from '../components/EventBanner';
+import Event from '../components/Event';
+import {
+  events,
+  SectionType,
+  EventGridStyle,
+  type EventItem,
+  type EventSection,
+} from '../config';
+
+const processEvents = (events: EventItem[]) => {
+  let mainEvent: EventItem | undefined = undefined;
+  const futureEvents: EventItem[] = [];
+  const currentEvents: EventItem[] = [];
+  const pastEvents: EventItem[] = [];
+  // Sort Events
+  events.sort((a, b) => a.date.getTime() - b.date.getTime());
+  // Process Events
+  for (const event of events) {
+    // Sanity TypeChecking
+    if (event.date < event.open_date)
+      throw new Error('ConfigError: Event opens after event date');
+    // handle main event
+    if (event.main_event) {
+      if (mainEvent == null) mainEvent = event;
+      else throw new Error('ConfigError: Multiple events marked as main');
```

Comment:
I am going to apply the zod change you suggested but I am also going to keep this error (I might change it from a `ConfigError` to an `Impossible` though as we should never hit it). The reason I want to validate this here as well is because typescript doesn't really provide good protections to catch this, so if we inject events elsewhere like I am currently doing with fake_events things would break without throwing an error.